blockindex/contractstaking/cache.go

```diff
@@ -341,7 +346,7 @@ func (s *contractStakingCache) getBucket(id uint64) (*Bucket, bool) {
 func (s *contractStakingCache) getAllBucketInfo() map[uint64]*bucketInfo {
 	m := make(map[uint64]*bucketInfo)
 	for k, v := range s.bucketInfoMap {
-		m[k] = v
+		m[k] = v.clone()
```

Comment:
think we can remove this func?

it is only used in `func (s *contractStakingCache) Buckets()`, we can directly use 
```
for id, bi := range s.bucketInfoMap {
		bt := s.mustGetBucketType(bi.TypeIndex)
		vb := assembleBucket(id, bi.clone(), bt, s.contractAddress) // just clone bi here
		vbs = append(vbs, vb)
	}
```

Reply:
fixed